crates/swc_ecma_compat_es2017/src/async_to_generator.rs

```diff
@@ -62,23 +63,34 @@ pub struct Config {
 struct AsyncToGenerator<C: Comments + Clone> {
     c: Config,
     comments: Option<C>,
+    in_subclass: bool,
     unresolved_ctxt: SyntaxContext,
 }
 
 struct Actual<C: Comments> {
     c: Config,
     comments: Option<C>,
 
+    in_subclass: bool,
+    hoister: FnEnvHoister,
+
     unresolved_ctxt: SyntaxContext,
     extra_stmts: Vec<Stmt>,
-    hoist_stmts: Vec<Stmt>,
 }
 
 #[swc_trace]
 #[fast_path(ShouldWork)]
 impl<C: Comments + Clone> VisitMut for AsyncToGenerator<C> {
     noop_visit_mut_type!(fail);
 
+    fn visit_mut_class(&mut self, c: &mut Class) {
+        if c.super_class.is_some() {
+            self.in_subclass = true;
+        }
+        c.visit_mut_children_with(self);
+        self.in_subclass = false;
```

Comment:
Is it fine to store `false` directly?
I think it can cause a problem for statements after a nested class